rts/idris_utf8.h

```diff
@@ -1,24 +1,38 @@
 #ifndef _IDRIS_UTF8
 #define _IDRIS_UTF8
 
+#include <stdint.h>
+#include <stdlib.h>
+
 /* Various functions for dealing with UTF8 encoding. These are probably
    not very efficient (and I'm (EB) making no guarantees about their
    correctness.) Nevertheless, they mean that we can treat Strings as
    UFT8. Patches welcome :). */
 
+typedef uint8_t utf8_byte;
+
+// Pascal-style strings with their lengths.
+typedef struct {
+    size_t size;
+    utf8_byte* str_data;
+} SizedString;
```

Comment:
Could save some indirection by appending str_data as a zero-length array. It won't be computable at compile time so lose sizeof, but the length is in there so easy to substitute.